video/graphics.h

```diff
@@ -626,7 +641,13 @@ void setClippingRect(Rect rect) {
 // Draw cursor
 //
 void drawCursor(Point p) {
-	canvas->swapRectangle(p.X, p.Y, p.X + fontW - 1, p.Y + fontH - 1);
+	if (textCursorActive()) {
+		if (cursorHStart < fontW && cursorHStart <= cursorHEnd && cursorVStart < fontH && cursorVStart <= cursorVEnd) {
```

Comment:
the text cursor drawing rules are such that the start row must be before the end row, and the cursor will always smaller than the font size.

this matches Acorn's implementation, extended to also support adjusting the horizontal size of the cursor